### PR TITLE
Fix phpstan errors

### PR DESCRIPTION
**Pull request by MrPetovan**
I ran the PHPStan (https://github.com/phpstan/phpstan) static analyzer tool on Friendica's code And I fixed 663 errors out of the 668 it found.

Most of them were about implicit requires (inlucde/dba.php mostly), but about 100 of them were undefined variables. I could found a fix for most of them, but there are 5 left I have no idea how to fix, I left `fixme` comments in the code, happy hunting!

### CODE REVIEW

File: src/Protocol/Diaspora.php [hunk 1]
```diff
@@ -2308,16 +2310,16 @@ private static function receiveRequestMakeFriend($importer, $contact)
...
 
 				$arr["object"] = self::constructNewFriendObject($contact);
 
 				$arr["last-child"] = 1;
 
-				$arr["allow_cid"] = $user[0]["allow_cid"];
-				$arr["allow_gid"] = $user[0]["allow_gid"];
-				$arr["deny_cid"]  = $user[0]["deny_cid"];
-				$arr["deny_gid"]  = $user[0]["deny_gid"];
+				$arr["allow_cid"] = $self[0]["allow_cid"];
```
Comment (annando) [inline src/Protocol/Diaspora.php]: ```$user``` is undefined - but ```$self``` doesn't seem to be an appropriate replacement. It should be the defaults for posts that are defined in the ```user``` table. Please add a call for the ```user``` entry here.

File: src/Protocol/Diaspora.php [hunk 2]
```diff
@@ -4065,6 +4067,7 @@ public static function sendProfile($uid, $recips = false)
 
 		foreach ($recips as $recip) {
 			logger("Send updated profile data for user ".$uid." to contact ".$recip["id"], LOGGER_DEBUG);
+			/// @fixme $profile isn't available here
 			self::buildAndTransmit($profile, $recip, "profile", $message, false, "", true);
```
Comment (annando) [inline src/Protocol/Diaspora.php]: It has to be the ```$owner``` that is fetched - for example - here: https://github.com/friendica/friendica/blob/develop/src/Worker/Delivery.php#L143-L155

Perhaps we should unify this query and use it here.

File: src/Protocol/OStatus.php [hunk 3]
```diff
@@ -1023,6 +1024,7 @@ private static function processRepeatedItem($xpath, $entry, &$item, $importer)
 		$orig_created = $xpath->query('atom:published/text()', $activityobjects)->item(0)->nodeValue;
 		$orig_edited = $xpath->query('atom:updated/text()', $activityobjects)->item(0)->nodeValue;
 
+		/// @fixme $contact is unavailable here
 		$orig_contact = $contact;
```
Comment (annando) [inline src/Protocol/OStatus.php]: Just remove this line and replace ```orig_contact``` in the following line with ```contact```. I guess this is some leftover from the time that this functionality wasn't in a dedicated function.

File: src/Protocol/PortableContact.php [hunk 4]
```diff
@@ -815,7 +816,8 @@ public static function detectServerType($body)
 			return false;
 		}
 
-		$server["site_name"] = $xpath->evaluate($element."//head/title/text()", $context)->item(0)->nodeValue;
+		/// @fixme $element is unavailable here
+		$server["site_name"] = $xpath->evaluate($element."//head/title/text()")->item(0)->nodeValue;
```
Comment (annando) [inline src/Protocol/PortableContact.php]: ```$element``` does not make sense at all. Just remove it.

File: src/Protocol/OStatus.php (hunk 3 above)
Comment (MrPetovan) [reply src/Protocol/OStatus.php]: The problem is that there's no `$contact` at all.

File: src/Worker/Directory.php [hunk 5]
```diff
@@ -25,6 +25,7 @@ public static function execute($url = '') {
 
 		$dir .= "/submit";
 
+		/// @fixme $argv is unavailable here
 		$arr = array('url' => $argv[1]);
```
Comment (annando) [inline src/Worker/Directory.php]: It's just ```$url``` I guess.

File: src/Worker/Queue.php [hunk 6]
```diff
@@ -54,7 +55,7 @@ public static function execute($queue_id = 0)
 			 */
 			$r = q("SELECT `id` FROM `queue` WHERE ((`created` > UTC_TIMESTAMP() - INTERVAL 12 HOUR AND `last` < UTC_TIMESTAMP() - INTERVAL 15 MINUTE) OR (`last` < UTC_TIMESTAMP() - INTERVAL 1 HOUR)) ORDER BY `cid`, `created`");
 
-			call_hooks('queue_predeliver', $a, $r);
```
Comment (annando) [inline src/Worker/Queue.php]: This is a really ancient bug.

File: src/Protocol/OStatus.php (hunk 3 above)
Comment (annando) [reply src/Protocol/OStatus.php]: This is no problem. It is called by reference.

File: src/Protocol/Diaspora.php [hunk 7]
```diff
@@ -4051,6 +4051,29 @@ public static function sendProfile($uid, $recips = false)
 			return;
 		}
 
+		$r = dba::p("SELECT
```
Comment (annando) [inline src/Protocol/Diaspora.php]: Wouldn't it be better to move this into the ```Contact``` class as method? We are using this very often.

File: src/Protocol/Diaspora.php [hunk 8]
```diff
@@ -4051,28 +4051,10 @@ public static function sendProfile($uid, $recips = false)
...
-			FROM `contact`
-			INNER JOIN `user`
-				ON `user`.`uid` = `contact`.`uid`
-			WHERE `contact`.`uid` = ?
-			AND `contact`.`self` = 1
-			LIMIT 1",
-			$uid
-		);
-		if (!DBM::is_result($r)) {
+		$owner = User::getOwnerDataById($uid);
```
Comment (annando) [inline src/Protocol/Diaspora.php]: Okay, you already did it ;-)

File: src/Protocol/Diaspora.php (hunk 7 above)
Comment (MrPetovan) [reply src/Protocol/Diaspora.php]: Your review is already outdated, and I put it in `User` because we're using a user id.

**Comment by MrPetovan** [comment]
So, a little bit of GitHub-fu for you @annando : You don't have to submit your review comment one by one. If you select "Add review comment", they are stored in Pending state, until you're done with the current review, and you can finally choose your review type (comment, approve, request changes) and hit the review button. It will publish all your comments at once and will prevent the kind of cross-post shenanigans I did with you now.

**Comment by annando** [comment]
@MrPetovan Yeah, I know. Don't ask me why I hadn't done that here.

**Comment by MrPetovan** [comment]
It's been a couple times already, I wasn't sure if you were aware of the review feature in general.

**Comment by annando** [comment]
Yeah, it's some kind of discipline that I don't have at all :-)

**Comment by annando** [comment]
I saw in other projects that they did some automatic check for every incoming PR. Do you think we could the same with this and similar scripts?

**Comment by MrPetovan** [comment]
I have to look into it but it sounds like it’s mostly configuration.

I tried https://github.com/vimeo/psalm as well but it is expecting PHP 7 only code so it barked every time a function wasn’t explicitly specifying the return type.

**Pull request merged**